src/web/mjs/engine/DownloadManager.mjs

```diff
@@ -52,4 +94,18 @@ export default class DownloadManager extends EventTarget {
             }
         }
     }
+
+    /**
+     * Clear the download queue.
+     */
+    clearQueue() {
+        for (let connectorID in this.queue) {
+            const jobs = this.queue[connectorID];
```

Comment:
Can this be simplified by using existing functionality?
`this.queue[connectorID].slice(0).forEach(this.removeFromQueue);`